webpack/components/QuestionResults.js

```diff
@@ -1,35 +1,99 @@
 import React, { Component } from 'react';
 import { questionProps } from "../prop-types/question_props";
 import { formProps } from "../prop-types/form_props";
+import { Link } from 'react-router';
 
 export default class QuestionResults extends Component {
   render() {
     return (
       <div className="question-group">
-        <div className="row">
-          <div className="col-md-1"><b>ID</b></div>
-          <div className="col-md-11"><b>Content</b></div>
-        </div><br/>
-        {this.props.questions.map((question, i) => {
+        <div>
+        {!(this.props.questions === undefined)  && this.props.questions.map((question, i) => {
           return (
             <div className="row" key={i}>
               <div data-question-id={question.id}>
-                <div className="col-md-8" name="question_content" >{question.content}</div>
-                <div className="col-md-3">
-                  <div id={"question_"+question.id+"_add"} className="btn btn-small btn-default"
-                       onClick={() => this.props.addQuestion(this.props.form, question)}>
-                    <b>Add</b>
-                  </div>
-                </div>
+                {this.questionResult(question)}
               </div>
             </div>
           );
         })}
+        </div>
+      </div>
+    );
+  }
+  reviseOrEditLink(question) {
```

Comment:
The way these buttons render should probably be unified with the logic used already in other places - possibly extracted into a higher level module - but I am not going to let anything regarding this or the search section be a blocker for right now since we will be changing this to the uber widget + elastic search search soon anyway.